back/src/routes/book-router.ts

```diff
@@ -0,0 +1,151 @@
+import { Router } from "express";
+import { ClubModel } from "../models/club";
+import { BookModel, BookInterface } from "../models/book";
+
+const bookRouter = Router();
+
+// get all
+bookRouter.get("/", async (req, res) => {
+  const books = await BookModel.find();
+  res.status(200).json({ books });
+});
+
+
+// get by id
+bookRouter.get("/:bookId", async (req, res) => {
+  const { bookId } = req.params;
+
+  if (!bookId) {
+    return res.status(400).json("Parameter is missing");
+  }
+
+  try {  
+    const book = await BookModel.findById({ _id: bookId });
+
+    if (!book) {
+      return res.status(404).json({ message: "Book not found" });
+    }
+
+    res.status(200).json(book);
+
+  } catch(err) {
+    console.error("Error getting book:", err);
+    res.status(500).json({ message: "Something went wrong. Please try again." });
+  }
+});
+
+
+// create book
+bookRouter.post("/", async (req, res) => {
+  const { title, description, author, pages, cover, startDate, endDate, clubId} = req.body;
+
+  if (!title || !description || !author || !pages || !cover || !startDate || !endDate || !clubId) {
+    return res.status(400).json({ message: "Missing required parameters. Please provide all fields." });
+  }
+
+  try {
+    const newBook = await BookModel.create({
+      title,
+      description,
+      author,
+      pages,
+      cover,
+      startDate,
+      endDate,
+      clubId,
+    });
+
+    const updatedClub = await ClubModel.findByIdAndUpdate(
+      clubId,
+      { $push: { books: newBook._id } },
+      { new: true }
+    );
+
+    if (!updatedClub) {
+      return res.status(404).json({ message: "Club not found. Unable to add book." });
+    }
+
+    return res.status(201).json({ message: "Book created successfully", book: newBook });
+
+  } catch (err) {
+    console.error("Error creating book:", err);
+    res.status(500).json({ message: "Something went wrong. Please try again." });  
+  }
+});
+
+
+// update info
+bookRouter.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const { title, description, author, pages, cover, startDate, endDate } = req.body;
+
+  if (!title && !description && !author && !pages && !cover && !startDate && !endDate) {
+    return res.status(400).json({ message: "Missing required parameters. Please provide all fields." });
+  }
+
+  try {
+
+    /// TODO: make a find by id func
+    const book = await BookModel.findById(id);
+    if (!book) {
+      return res.status(404).json({ message: "Book not found" });
+    }
+
+    const updateFields: Partial<BookInterface> = {};
+    if (title) updateFields.title = title;
```

Comment:
I wonder if mogoose gets undefined it knows to ignore it. and if we can define in the schema that a value is not nullable then it throws an error
if so we can jest send the object for save